CDP4Common/Poco/NestedElement.cs

```diff
@@ -74,5 +74,82 @@ protected DomainOfExpertise GetDerivedOwner()
         /// is the root of the Nested tree.
         /// </summary>
         public bool IsRootElement { get; internal set; }
+
+        /// <summary>
+        /// The <see cref="ElementBase"/> representing a <see cref="NestedElement"/>.
+        /// </summary>
+        /// <returns>
+        /// The <see cref="ElementBase"/>.
+        /// </returns>
+        public ElementBase GetElementBase()
+        {
+            return this.IsRootElement
+                ? (ElementBase)this.RootElement
+                : this.ElementUsage.Last();
+        }
+
+        /// <summary>
+        /// The <see cref="ElementDefinition"/> representing a <see cref="NestedElement"/>, if it exists.
+        /// </summary>
+        /// <returns>
+        /// The <see cref="ElementDefinition"/>.
+        /// </returns>
+        public ElementDefinition GetElementDefinition()
+        {
+            var elementBase = this.GetElementBase();
+
+            return elementBase as ElementDefinition ?? (elementBase as ElementUsage)?.ElementDefinition;
+        }
+
+        /// <summary>
+        /// The <see cref="ElementUsage"/> representing a <see cref="NestedElement"/>, if it exists.
+        /// </summary>
+        /// <returns>
+        /// The <see cref="ElementUsage"/>.
+        /// </returns>
+        public ElementUsage GetElementUsage()
+        {
+            return this.GetElementBase() as ElementUsage;
+        }
+
+        /// <summary>
+        /// Checks if a <see cref="NestedElement"/> contains a specific <see cref="Category"/>.
```

Comment:
this should be rephrased to:

Asserts whether the NestedElelement is a member of the specific Category

here i am wondering whether super categories are taken into account. There are helper methods for that as well in the SDK, i think extension methods on the Category class

Reply:
Yes, they are on the ICategorizableThing. Refactored it.